### ISSUE TITLE
Exam: Points displayed to student randomly differ from points displayed to instructor - review impossible

### ISSUE THREAD

**Issue by mitcoef** [ls1intum/Artemis#7131]
### Describe the bug

When reviewing the exam, students may see a different number of points displayed for an exercise than the instructor assessing the complaint. This difference does not correspond to first assessment - as far as we can tell, it's random.

### To Reproduce

We are currently unable to reproduce. We do have access to links where the behavior is demonstrated. Since this is from an actual exam, we hesitate to share for reasons of data protection. See screenshots below.

### Expected behavior

Points displayed to student must be identical to points displayed to instructor as well as actual points given for the exercise.
Otherwise, a review is simply not possible. 

### Screenshots
**On instructor side, 5.5 points given:**

![image](https://github.com/ls1intum/Artemis/assets/36709930/0a224ac6-f847-4001-972b-a8bd196e3507)

**On student side, 4.5 points given:**

![image](https://github.com/ls1intum/Artemis/assets/36709930/3b2fe7ad-f065-4af4-bc51-59f9ece5083f)

**First assessment, 3.5 points given:**

![image](https://github.com/ls1intum/Artemis/assets/36709930/dc00ee84-fd69-4b41-bcbb-7dfb7cb3617f)

### Which version of Artemis are you seeing the problem on?

6.4.1

### What browsers are you seeing the problem on?

Firefox

### Additional context

The complaint seen in the example screenshots was viewed by multiple instructors on different accounts and machines. The issue persisted in all cases.

**Comment by Strohgelaender** [comment]
Can you share a bit more information on which scores where given when:
- What was the socore in the first corrrection round? How did it chage after the second correction round?
- Does the student see a mixture of first and second round results at different places?
- After submitting the complaint, which result gets used as a basis?

The issue description contains 3 different results (5.5 points, 4.5 points, and 3.5 points), where exactly is which result visible?

**Comment by pal03377** [comment]
If you remove the `feedback.reference` in the `feedback` table, the student can see the feedback, but the instructor cannot. The error would also occur if the `reference` points to a non-existing `TextBlock`. This probably isn't included in the calculation because it is neither referenced feedback (`reference` is missing) nor unreferenced feedback (`type != FeedbackType.MANUAL_UNREFERENCED`).

Now, the client code gets a bit weird. When creating a `TextBlock`, you must call `textBlock.computeId()` so that the `TextBlock` has an ID. This means we must remember to call this at all possible places in the code. This probably gets forgotten somewhere, and that's how it breaks. For example, this could be the case with custom TextBlocks (which you create with Opt/Alt). However, I couldn't reproduce it with that either. The ID from `computeId()` is a hash of submissionID, startIndex, endIndex, and text.

Because I also refactored some things in this regard in #6861, it's possible that the error no longer occurs at all, and therefore, we can't reproduce it :sweat_smile:

Unfortunately, there seems to be no simple solution for the specific problem of this issue: We would have to fix the "broken" feedback in the database. It is probably Text Exercise Feedback with `reference=NULL` and `type=MANUAL` (= 0):

```sql
SELECT * 
FROM feedback f JOIN result r ON f.result_id=r.id JOIN participation p on r.participation_id=p.id JOIN exercise e on p.exercise_id=e.id
WHERE e.discriminator='T'/*ext exercise*/ AND f.reference IS NULL AND f.type=/*manual*/0;
```

Here is a PR that might prevent such issues in the future: #7134

**Comment by krusche** [comment]
#7134 might fix the issue in the soon to be released version 6.4.2
In case you encounter this issue again, please reopen it and provide more details so we can reproduce it. Thank you

**Comment by krusche** [comment]
> ```sql
> SELECT * 
> FROM feedback f JOIN result r ON f.result_id=r.id JOIN participation p on r.participation_id=p.id JOIN exercise e on p.exercise_id=e.id
> WHERE e.discriminator='T'/*ext exercise*/ AND f.reference IS NULL AND f.type=/*manual*/0;
> ```

I tried to execute this query, however it is too slow. We have ~62 mio feedback rows in the production db and the column `type` does not have an index.
